### PR TITLE
Add all files required for get conversations of a user

### PR DESCRIPTION
**Pull request by enzofab91**
### **Feature:**
As a user I should be able to see all my conversations.

### **Trello reference:**
https://trello.com/c/gCj9IZXI

### **Description:**
This PR add all files required for creating a user conversation when a match between two users is created. For the time being, only index endpoint is supported.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -494,6 +494,45 @@ FORMAT: 1A
...
+
+    + Body
+
++ Response 200
+
+    + Body
+
+            "conversations": [
+                {
+                  "conversations": {
```
Comment (mrodriguez90) [inline apiary.apib:528]: why is the conversation object inside another object? Just a question because it's not clear to me.

Comment (enzofab91) [reply apiary.apib:528]: Copy & Paste error.

File: .rubocop.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ Layout/LineLength:
     - db/seeds.rb
     - spec/factories/users.rb
     - spec/requests/api/v1/**/*
+    - spec/models//**/*
```
Comment (martinjaimem) [inline .rubocop.yml:26]: does this work?
shouldn't it be sth like `- spec/models/**/*`?

Comment (enzofab91) [reply .rubocop.yml:26]: It works. It seems that ignore the extra slash. BTW, now its fixed.

File: app/forms/match_form.rb [hunk 3]
```diff
@@ -28,4 +29,8 @@ def notify_match
 
     FcmService.sent_notification(registration_ids, I18n.t('api.notifications.new_match'), data)
   end
+
+  def create_conversation
+    Conversation.create!(Hash(match_id: match.id))
```
Comment (martinjaimem) [inline app/forms/match_form.rb:34]: doesn't `Conversation.create!(match_id: match.id)` work?

File: app/models/conversation.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    users << [match.first_user, match.second_user]
+  end
+
+  def users_uniqueness
+    first_user_conversations = UserConversation.where(user_id: match.first_user_id)
+                                               .pluck(:conversation_id)
+    second_user_conversations = UserConversation.where(user_id: match.second_user_id)
+                                                .pluck(:conversation_id)
+
+    # if both conversations intersections match means already exist a conversation between them
```
Comment (martinjaimem) [inline app/models/conversation.rb:40]:
```suggestion
```

File: spec/models/user_conversation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+#  index_user_conversations_on_user_id          (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (conversation_id => conversations.id)
+#  fk_rails_...  (user_id => users.id)
+#
+require 'rails_helper'
+
+RSpec.describe UserConversation, type: :model do
```
Comment (martinjaimem) [inline spec/models/user_conversation_spec.rb:23]:
```suggestion
describe UserConversation, type: :model do
```

File: spec/requests/api/v1/conversations/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+describe 'GET api/v1/conversations', type: :request do
+  let(:target) { create :target }
```
Comment (martinjaimem) [inline spec/requests/api/v1/conversations/index_spec.rb:2]:
```suggestion
  let(:target) { create(:target) }
```

File: spec/models/conversation_spec.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+#
+# Indexes
+#
+#  index_conversations_on_match_id  (match_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (match_id => matches.id)
+#
+require 'rails_helper'
```
Comment (martinjaimem) [inline spec/models/conversation_spec.rb:18]:
```suggestion
```

File: spec/models/conversation_spec.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+RSpec.describe Conversation, type: :model do
```
Comment (martinjaimem) [inline spec/models/conversation_spec.rb:20]:
```suggestion
describe Conversation, type: :model do
```

File: apiary.apib [hunk 9]
```diff
@@ -494,6 +494,45 @@ FORMAT: 1A
...
+
+            "conversations": [
+                {
+                  "conversation": {
+                    "id": 1,
+                    "match_id": "match_id"
+                  }
+                }
+              ]
+            }
```
Comment (martinjaimem) [inline apiary.apib:534]:
```suggestion
            "conversations": [
                {
                  "conversation": {
                    "id": 1,
                    "match_id": "match_id"
                  }
                }
              ]
```

**Comment by martinjaimem** [comment]
Niceee 😎

File: spec/requests/api/v1/conversations/index_spec.rb [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  subject { get api_v1_conversations_path, headers: auth_headers, as: :json }
+
+  it 'returns a successful response' do
+    subject
+    expect(response).to be_successful
+  end
+
+  it 'returns user conversations' do
+    subject
+    expect(json[:conversations].pluck([:id])).to match_array(conversations.pluck([:id]))
```
[the remark this answers is not in the document]
Comment (enzofab91) [reply spec/requests/api/v1/conversations/index_spec.rb:18]: Doen't work.

**Review by sebastiancaraballo — APPROVED** [review]

File: app/models/conversation.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (enzofab91) [reply app/models/conversation.rb:40]: When a conversation is created, it generates two records in user_conversations, one for each user in the conversation.
To check uniqueness conversation between two users I have to check both records exists for one conversation.

**Pull request merged**